### DISCUSSION TITLE
Project Idea

### DISCUSSION THREAD

**Discussion by shroffk** [epics-base/epics-base#462] [Codeathon 2024]
Looking for project ideas including suggestions for improved documentation for
 - EPICS core
 - Phoebus Tools
 - Middle layer services

**Comment by daykin** [comment]
If it doesn't already exist, I would like to create a graphical click-tracking/analytics module for Phoebus. I want to see what tasks our operators are performing regularly, and improve locality between the most common tasks.

**Comment by jeonghanlee** [comment]
### Java build systems

We need to address the challenges posed by different Java build systems such as Ant, Maven, and Gradle, especially with respect to backward compatibility issues. This is critical to ensure smooth transitions and compatibility across projects and releases.

### Pull Request Documentation and Code Review

There's a noticeable lack of comments and documentation in many pull requests, especially in repositories like phoebus, epicsarchiverep, and channelfinder. While the core team may handle tasks internally, it's important to establish more formal processes for sharing and documenting changes. This provides transparency and facilitates smoother collaboration among team members.

#### To improve pull request quality, we could implement:

- **Templates**: Provide pre-defined templates for pull requests that outline expected documentation sections like `Changes Made`, `Testing Performed`, and `Potential Impacts`.
- **Code Review Checklist**: Create a checklist for code reviewers to ensure key areas like code clarity, adherence to coding standards, and unit test coverage are addressed.


### Update the base EPICS templates

We should prioritize updating the base EPICS templates, specifically to include support for PVAccess in `makeBaseApp/top/iocApp/src/Makefile`. This update is critical to improve functionality and ensure compatibility with modern tools and technologies.

Updating the templates with PVAccess support will allow for:

- **Enhanced Functionality**: Utilize the latest features and capabilities offered by PVAccess.
- **Improved Efficiency**: Streamline workflows and potentially reduce development time by leveraging PVAccess functionalities.


### iocStats status

We need to review the current state of iocStats and ensure that it meets the project goals and requirements. This includes assessing

**Reply by hjunkes** [reply]
Hi Han,
I would collaborate on the two points "EPICS template" and "iocStats". These two points are current work in progress and important for us (FHI).

**Reply by exzombie** [reply]
If the templates are being updated, I would like to use the opportunity to clarify the licensing of the template files and add SPDX tags.

**Comment by hanlet** [comment]
I'm interested in exploring the possibility of developing a generic driver for SoM devices.  If the FPGA developer agrees to use and can provide the driver a memory map and type in userIO space, then the driver could simply read/write normative types from/to these addresses.  That part appears straight forward, but I/O interrupts are also required.
